zcash_client_backend/src/data_api/chain.rs

```diff
@@ -479,142 +472,167 @@ impl ScanSummary {
 /// This function will return after scanning at most `limit` new blocks, to enable the caller to
 /// update their UI with scanning progress. Repeatedly calling this function with `from_height ==
 /// None` will process sequential ranges of blocks.
-#[tracing::instrument(skip(params, block_source, data_db))]
+#[tracing::instrument(skip(params, block_cache, wallet_data))]
 #[allow(clippy::type_complexity)]
-pub fn scan_cached_blocks<ParamsT, DbT, BlockSourceT>(
+pub fn scan_cached_blocks<ParamsT, DbT, BcT>(
     params: &ParamsT,
-    block_source: &BlockSourceT,
-    data_db: &mut DbT,
-    from_height: BlockHeight,
-    limit: usize,
-) -> Result<ScanSummary, Error<DbT::Error, BlockSourceT::Error>>
+    block_cache: &BcT,
+    wallet_data: &mut DbT,
+    scan_range: &ScanRange,
+) -> Result<ScanSummary, Error<DbT::Error, BcT::Error>>
```

Comment:
As a suggestion for future PRs: part of the reason why this PR is harder to rebase is because there are commits like this one that rename all of the arguments as well as changing the parameter name, which were not necessary to do. In particular, renaming `data_db` to `wallet_data` in the same commit as switching from `BlockSource` to `BlockCache`.